### DISCUSSION TITLE
Autodcomplete 组件怎么监听input失去焦点？

### DISCUSSION THREAD

**Discussion by peterzhou84** [element-plus/element-plus#10950] [Q&A]
Autodcomplete 组件目前只暴露了select和change事件，怎么才能监听里面的input组件的blur事件呢？

**Comment by chenxch** [comment]
可以从文档暴露中找到以下两个属性
![image](https://user-images.githubusercontent.com/23251408/206841109-6e9a3ef5-5683-45f2-9323-d608e30b56d4.png)
可以通过el.value.inputRef.input找到input的dom，那就可以利用原生的addEventListener添加 blur事件
```ts
onMounted(() => {
  restaurants.value = loadAll()
  console.log(el.value.inputRef.input)
  el.value.inputRef.input.addEventListener('blur',onBlur)
})
```
然后考虑到联想框的点击也会导致input的blur触发，当时组件是会保持这个focus，所以需要根据activated进行判断
```ts
function onBlur(){
  if(el.value?.activated){
    return
  }
  console.log('blur')
}
```
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtcm93IGNsYXNzPVwiZGVtby1hdXRvY29tcGxldGVcIj5cbiAgICA8ZWwtY29sIDpzcGFuPVwiMTJcIj5cbiAgICAgIDxkaXYgY2xhc3M9XCJzdWItdGl0bGUgbXktMiB0ZXh0LXNtIHRleHQtZ3JheS02MDBcIj5cbiAgICAgICAgbGlzdCBzdWdnZXN0aW9ucyB3aGVuIGFjdGl2YXRlZFxuICAgICAgPC9kaXY+XG4gICAgICA8ZWwtYXV0b2NvbXBsZXRlXG4gICAgICAgIHJlZj1cImVsXCJcbiAgICAgICAgdi1tb2RlbD1cInN0YXRlMVwiXG4gICAgICAgIDpmZXRjaC1zdWdnZXN0aW9ucz1cInF1ZXJ5U2VhcmNoXCJcbiAgICAgICAgY2xlYXJhYmxlXG4gICAgICAgIGNsYXNzPVwiaW5saW5lLWlucHV0IHctNTBcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBJbnB1dFwiXG4gICAgICAgIEBzZWxlY3Q9XCJoYW5kbGVTZWxlY3RcIlxuICAgICAgLz5cbiAgICA8L2VsLWNvbD5cblxuICA8L2VsLXJvdz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyBvbk1vdW50ZWQsIHJlZiB9IGZyb20gJ3Z1ZSdcblxuaW50ZXJmYWNlIFJlc3RhdXJhbnRJdGVtIHtcbiAgdmFsdWU6IHN0cmluZ1xuICBsaW5rOiBzdHJpbmdcbn1cblxuY29uc3Qgc3RhdGUxID0gcmVmKCcnKVxuY29uc3Qgc3RhdGUyID0gcmVmKCcnKVxuY29uc3QgZWwgPSByZWYobnVsbClcblxuY29uc3QgcmVzdGF1cmFudHMgPSByZWY8UmVzdGF1cmFudEl0ZW1bXT4oW10pXG5jb25zdCBxdWVyeVNlYXJjaCA9IChxdWVyeVN0cmluZzogc3RyaW5nLCBjYjogYW55KSA9PiB7XG4gIGNvbnN0IHJlc3VsdHMgPSBxdWVyeVN0cmluZ1xuICAgID8gcmVzdGF1cmFudHMudmFsdWUuZmlsdGVyKGNyZWF0ZUZpbHRlcihxdWVyeVN0cmluZykpXG4gICAgOiByZXN0YXVyYW50cy52YWx1ZVxuICAvLyBjYWxsIGNhbGxiYWNrIGZ1bmN0aW9uIHRvIHJldHVybiBzdWdnZXN0aW9uc1xuICBjYihyZXN1bHRzKVxufVxuY29uc3QgY3JlYXRlRmlsdGVyID0gKHF1ZXJ5U3RyaW5nOiBzdHJpbmcpID0+IHtcbiAgcmV0dXJuIChyZXN0YXVyYW50OiBSZXN0YXVyYW50SXRlbSkgPT4ge1xuICAgIHJldHVybiAoXG4gICAgICByZXN0YXVyYW50LnZhbHVlLnRvTG93ZXJDYXNlKCkuaW5kZXhPZihxdWVyeVN0cmluZy50b0xvd2VyQ2FzZSgpKSA9PT0gMFxuICAgIClcbiAgfVxufVxubGV0IHNlbGVjdEZsYWcgPSBmYWxzZVxuZnVuY3Rpb24gb25CbHVyKCl7XG4gIGlmKGVsLnZhbHVlPy5hY3RpdmF0ZWQpe1xuICAgIHJldHVyblxuICB9XG4gIGNvbnNvbGUubG9nKCdibHVyJylcbn1cbmNvbnN0IGxvYWRBbGwgPSAoKSA9PiB7XG4gIHJldHVybiBbXG4gICAgeyB2YWx1ZTogJ3Z1ZScsIGxpbms6ICdodHRwczovL2dpdGh1Yi5jb20vdnVlanMvdnVlJyB9LFxuICAgIHsgdmFsdWU6ICdlbGVtZW50JywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS9FbGVtZUZFL2VsZW1lbnQnIH0sXG4gICAgeyB2YWx1ZTogJ2Nvb2tpbmcnLCBsaW5rOiAnaHR0cHM6Ly9naXRodWIuY29tL0VsZW1lRkUvY29va2luZycgfSxcbiAgICB7IHZhbHVlOiAnbWludC11aScsIGxpbms6ICdodHRwczovL2dpdGh1Yi5jb20vRWxlbWVGRS9taW50LXVpJyB9LFxuICAgIHsgdmFsdWU6ICd2dWV4JywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS92dWVqcy92dWV4JyB9LFxuICAgIHsgdmFsdWU6ICd2dWUtcm91dGVyJywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS92dWVqcy92dWUtcm91dGVyJyB9LFxuICAgIHsgdmFsdWU6ICdiYWJlbCcsIGxpbms6ICdodHRwczovL2dpdGh1Yi5jb20vYmFiZWwvYmFiZWwnIH0sXG4gIF1cbn1cblxuY29uc3QgaGFuZGxlU2VsZWN0ID0gKGl0ZW06IFJlc3RhdXJhbnRJdGVtKSA9PiB7XG4gIHNlbGVjdEZsYWcgPSB0cnVlXG4gIGNvbnNvbGUubG9nKGl0ZW0pXG59XG5cbm9uTW91bnRlZCgoKSA9PiB7XG4gIHJlc3RhdXJhbnRzLnZhbHVlID0gbG9hZEFsbCgpXG4gIGNvbnNvbGUubG9nKGVsLnZhbHVlLmlucHV0UmVmLmlucHV0KVxuICBlbC52YWx1ZS5pbnB1dFJlZi5pbnB1dC5hZGRFdmVudExpc3RlbmVyKCdibHVyJyxvbkJsdXIpXG59KVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)
